### ISSUE TITLE
Triple Strength Myomer

### ISSUE THREAD

**Issue by Aliencreature** [CptMoore/MechEngineer#68]
@CptMoore mentioned it in the engine upgrade thread, and because it seems very mechanically similar to your current work on engine crit handling, I wanted to get your opinion on this.

TSM occupies 6 slots and functions according to heat level. If a mech with TSM ends its turn with heat > 9, the next turn it receives:

1) +2 Walking MP, and recalculated Running MP based on that
2) 2x Melee Damage

**What Can Be Done Right Now**
1) Implement TSM as a third weight-saving type so that it can be mounted separately from armor or structure, but fall under the scrutiny of the required slot checks.
2) Add a TSM item that gives the above statusEffects permanently.

**What Is Required**
1) After the work on engine critical states has completed, the code that triggers/removes the statusEffects can be copied to do something similar based on heat level instead of crits. The statusEffect would define the required heat level in some unused property and the code would apply the statusEffect when that heat level is reached.

**What Could Be Done Further**
1) Implement a similar system to enable and disable stealth armor. One of the stealth armor parts (Stealth Armor Switch?) would function as a 'support weapon' with extreme range and zero/near zero damage. On firing, it would multiply a custom statistic with a default value of -1 by -1 through a statusEffect, and whether or not the statusEffects of the actual stealth armor are active would be decided by that statistic. -1 = inactive, +1 = active.

Thoughts?

**Comment by CptMoore** [comment]
> heat level instead of crits

Ill have to think about it when implementing generic crits as this requires yet another trigger type.

Also have to make weight savings more abstract, also because of the ongoing work for armor. probably will do a restrict placement parent class.

so what you say about triple strengh all makes sense.

only issue i see is that roguetech already has some melee (and masc) mods, do we need to implement them too?

> stealth armor

i feel we need a proper sensors mod first to make this useful. having a fake component to switch on/off sounds like a reasonable workaround before we can mod the skills bar directly.

> lots of features

maybe we should decide milestones and phases on what features we want to concentrate on once we get all these ideas collected.

**Comment by Aliencreature** [comment]
Yup, this is a low priority item, obviously. It was the MechEngineMod at first, the actual engines should be worked out first before we succumb to feature creep.

The difference between this and RogueTech's implementation is that RogueTech does for TSM the same fudging I did for stealth armor: in exchange for giving its effect at all times, it gets reduced bonuses from its Tabletop version. I also did something similar by making stealth armor be more forgiving with its heat penalty because the player can't shut it off to cool down, our version adds +10 heat/turn instead of the +30 mandated by TT's rules.

It's a nice thing to have, but not a must, and definitely not anything to do more than theorycraft on until the first release.
